weatherflow2mqtt/weatherflow_mqtt.py

```diff
@@ -166,19 +176,346 @@ async def connect(self) -> None:
+            )
+            high_low_data = self.sql.readHighLow()
+            self._add_to_queue(
+                highlow_topic, json.dumps(high_low_data), qos=1, retain=True
+            )
+            self.high_low_last_run = datetime.now().timestamp()
+
+        # Update the Forecast if it is time and enabled
+        if (
+            self.forecast is not None
+            and ((now := datetime.now().timestamp()) - self.forecast_last_run)
+            >= self.forecast.interval * 60
+        ):
+            fcst_state_topic = MQTT_TOPIC_FORMAT.format(
+                DOMAIN, FORECAST_ENTITY, "state"
+            )
+            fcst_attr_topic = MQTT_TOPIC_FORMAT.format(
+                DOMAIN, FORECAST_ENTITY, "attributes"
+            )
+            condition_data, fcst_data = await self.forecast.update_forecast()
+            if condition_data is not None:
+                self._add_to_queue(fcst_state_topic, json.dumps(condition_data))
+                self._add_to_queue(fcst_attr_topic, json.dumps(fcst_data))
+            self.forecast_last_run = now
+
+    def _add_to_queue(
+        self, topic: str, payload: str | None = None, qos: int = 0, retain: bool = False
+    ) -> None:
+        """Add an item to the queue."""
+        self._queue.put_nowait((topic, payload, qos, retain))
+
+    def _device_discovered(self, device: WeatherFlowDevice) -> None:
+        """Handle a discovered device."""
+
+        def _load_complete():
+            print(device, "was loaded")
+            self._setup_sensors(device)
+            device.on(
+                EVENT_STATUS_UPDATE,
+                lambda event: self._handle_status_update_event(device, event),
+            )
+            if isinstance(device, WeatherFlowSensorDevice):
+                device.on(
+                    EVENT_OBSERVATION,
+                    lambda event: self._handle_observation_event(device, event),
+                )
+                if isinstance(device, AirSensorType):
+                    device.on(
+                        EVENT_STRIKE,
+                        lambda event: self._handle_strike_event(device, event),
+                    )
+                if isinstance(device, SkySensorType):
+                    device.on(
+                        EVENT_RAPID_WIND,
+                        lambda event: self._handle_wind_event(device, event),
+                    )
+                    device.on(
+                        EVENT_RAIN_START,
+                        lambda event: self._handle_rain_start_event(device, event),
+                    )
+
+        device.on(EVENT_LOAD_COMPLETE, lambda _: _load_complete())
+
+    def _get_sensor_payload(
+        self,
+        sensor: BaseSensorDescription,
+        serial_number: str,
+        state_topic: str,
+        attr_topic: str,
+        firmware: str,
+    ) -> OrderedDict:
+        """Construct and return a sensor payload."""
+        payload = OrderedDict()
+
+        payload["name"] = f"WF {sensor.name}"
```

Comment:
Do we need to prefix sensor names with `WF `? I left it in because my main goal was implementing the pyweatherflowudp package. So most all of the changes were related in getting that to work well while trying to keep a lot of other things the same.

Reply:
Yes I think we still need that. First of all, it would be a major breaking change if we took it out, and secondly, there are so many Weather integrations out there, that separating where the data comes from, is very useful.

Reply:
Maybe we could change it to be the device name/serial instead of "WF " to match other HA component standards in the future.

The only reason I suggest this is because the HA frontend will hide the first part of the sensor name that matches the device name. So if a device is named "Hub HB-0001" and the sensor is named "Hub HB-0001 Status", it will display as just "Status" on the Hub's device page, which is nice since it removes the irrelevant/duplicative info to help you see what you really want to.

But for now I'll leave it.

Reply:
I like your idea. Leave as is for now, and then I will put a *deprecated* message in the release notes. It can then be changed on the next release.